runners/direct-java/src/main/java/org/apache/beam/runners/direct/TransformEvaluatorRegistry.java

```diff
@@ -77,15 +77,33 @@ private TransformEvaluatorRegistry(
   }
 
   @Override
+  public Collection<CommittedBundle<?>> getInitialInputs(AppliedPTransform<?, ?, ?> transform) {
```

Comment:
This is an example where splitting the two interfaces might be wise so that you have a mapping from certain root ptransforms to evaluators for roots, and this mapping for the non-root ones.